frontend/src/actions/IndependentWidgetsActions.js

```diff
@@ -0,0 +1,259 @@
+        });
+      }
+    );
+  };
+}
+
+/*
+ * @method patchViewAttributes
+ * @summary Patch changes to field
+ *
+ * @param {number} windowId
+ * @param {string} viewId
+ * @param {string} rowId
+ * @param {string} prop - field name
+ * @param {any} value - field value
+ */
+export function patchViewAttributes({ windowId, viewId, rowId, prop, value }) {
+  return (dispatch) => {
+    return patchViewAttributesRequest(
+      windowId,
+      viewId,
+      rowId,
+      prop,
+      value
+    ).then(({ data }) => {
+      if (data.length) {
+        dispatch({
+          type: PATCH_ATTRIBUTES,
+          payload: { data: data[0].fieldsByName },
+        });
+      }
+    });
+  };
+}
+
+/*
+ * @method setViewAttributesData
+ * @summary change local value of an attribute's field
+ *
+ * @param {string} field - field name
+ * @param {any} value
+ */
+export function setViewAttributesData({ field, value }) {
+  return {
+    type: SET_ATTRIBUTES_DATA,
+    payload: { field, value },
+  };
+}
+
+/*
+ * @method deleteViewAttributes
+ * @summary delete attributes data
+ */
+export function deleteViewAttributes() {
+  return {
+    type: DELETE_ATTRIBUTES,
+  };
+}
+
+//
+// TABLE QUICK INPUT
+//
+
+/**
+ * @method deleteQuickInput
+ * @summary Remove quick input's data
+ */
+export function deleteQuickInput() {
+  return {
+    type: DELETE_QUICKINPUT,
+  };
+}
+
+/**
+ * @method setQuickinputPending
+ * @summary Set the `inProgress` flag of QuickInput to true
+ */
+export function setQuickinputPending() {
+  return {
+    type: PATCH_QUICKINPUT_PENDING,
+  };
+}
+
+/**
+ * @method setQuickinputDone
+ * @summary Set the `inProgress` flag of QuickInput to false
+ */
+export function setQuickinputDone() {
+  return {
+    type: PATCH_QUICKINPUT_DONE,
+  };
+}
+
+/**
+ * @method setQuickinputData
+ * @summary Save field changes locally
+ *
+ * @param {object} fieldData - object with field names as keys and their data as values
+ */
+export function setQuickinputData(fieldData) {
+  return {
+    type: SET_QUICKINPUT_DATA,
+    payload: { fieldData },
+  };
+}
+
+/*
+ * @method patchQuickInput
+ * @summary Save QuickInput changes on the server
+ *
+ * @param {number} windowId
+ * @param {string} docId
+ * @param {string} tabId
+ * @param {string} prop
+ * @param {any} value
+ */
+export function patchQuickInput({ windowId, docId, tabId, prop, value }) {
+  return (dispatch, getState) => {
+    const state = getState();
+    const id = state.widgetHandler.quickInput.id;
+
+    dispatch(setQuickinputPending());
+
+    return patchRequest({
+      entity: 'window',
+      docType: windowId,
+      docId,
+      tabId,
+      property: prop,
+      value,
+      subentity: 'quickInput',
+      subentityId: id,
+    }).then(({ data }) => {
+      const fields = data[0] && data[0].fieldsByName;
+
+      if (fields) {
+        dispatch(setQuickinputData(fields));
+      }
+
+      dispatch(setQuickinputDone());
+    });
```

Comment:
.catch(...)   {  }
We need a way to say to the user that something went wrong. Use the message returned by the BE and render it in the err notification box we are using.

Reply:
We already do, because this Promise is handled by `completeRequest`:

![Screenshot 2021-04-19 at 16 10 14](https://user-images.githubusercontent.com/513460/115250547-f7ea7180-a129-11eb-9fde-f6249de9365a.png)

Reply:
We already do, because this Promise is handled by `completeRequest`:

![Screenshot 2021-04-19 at 16 10 14](https://user-images.githubusercontent.com/513460/115250547-f7ea7180-a129-11eb-9fde-f6249de9365a.png)